pyteal/ast/subroutine.py

```diff
@@ -611,6 +611,23 @@ def method_signature(self, overriding_name: str = None) -> str:
             overriding_name = self.name()
         return f"{overriding_name}({','.join(args)}){self.type_of()}"
 
+    def method_spec(self) -> dict:
+        skip_names = ["return", "output"]
+        args = [
+            {
+                "type": str(abi.type_spec_from_annotation(val)),
+                "name": name,
+            }
+            for name, val in self.subroutine.annotations.items()
+            if name not in skip_names
+        ]
+        return {
+            "name": self.name(),
+            "args": args,
+            "returns": {"type": str(self.type_of())},
+            "desc": "todo",  # Allow decorator to accept desc? Repurpose doc string?
```

Comment:
what do?

Reply:
read the doc string from the python function?

Reply:
We could have the best of both worlds for that comment - if it's not too much work
Eg, consider:

```python
@router.method
def foo():
    """
    This is a wonderful method
    """
    ...
```
It could generate

```python
    "description": "This is a wonderful method"
```

and similarly 

```python
@router.method(decription="This is a wonderful method")
def foo():
    ...
```

would generate the same. I guess we'd need to decide what do about

```python
@router.method(decription="This is a wonderful method")
def foo():
    "this is a terrible method - don't call it!"
    ...
```

As a sidenote, I advocate using `description` as opposed to `desc`

Reply:
i think if specified in decorator it should override doc string

noted about the desc/description 👍

Reply:
> i think if specified in decorator it should override doc string

That makes sense.

Reply:
@tzaffi `description` doesnt appear to work: https://github.com/algorand/py-algorand-sdk/blob/master/algosdk/abi/method.py#L212

Reply:
Since Abi hasn't been merged I advocate for changing desc to description everywhere in ABI

Reply:
Looks like its the same  `desc` across all of these, unless we have a really good reason to change it across all the SDKs I'm inclined to leave it as is

https://github.com/algorand/go-algorand-sdk/blob/develop/abi/interactions.go#L140
https://github.com/algorand/js-algorand-sdk/blob/develop/src/abi/method.ts#L149
https://github.com/algorand/java-algorand-sdk/blob/develop/src/main/java/com/algorand/algosdk/abi/Method.java#L80

Reply:
NVM. I see `desc` is built-in to ARC-4. Resolving.